src/clic.adoc

```diff
@@ -639,22 +571,34 @@ where:
 
 Synchronous exception traps always jump to OBASE.
 
-If the hart is currently running at some privilege mode (`pp`) at some
-interrupt level (`pil`) and an enabled interrupt becomes pending at
-any interrupt level in a higher privilege mode or if an interrupt assigned to a
-higher interrupt level in the current privilege mode becomes pending
-and interrupts are globally enabled in this privilege mode, then
-execution is immediately transferred to a handler running with the new
-interrupt's privilege mode (`**__x__**`) and interrupt level (`il`).
+==== New Interrupt Preemption Configuration ({mipreemptcfg}) CSR
+
+[source]
+----
+mipreemptcfg fields
+ bits        description
+ XLEN-1:4    (reserved)
+ 3:1         preemptmsk
+----
+
+{mipreemptcfg} is a WARL register used to configure the behavior of preemption.
+{mipreemptcfg}.{preemptmsk} configures the number of least significant bits of iprio masked out for constructing {ipprio}.
+The highest legal value of {mipreemptcfg}.{preemptmsk} is IPRIOLEN.
+The interrupt preemption priority {ipprio} is defined as follows:
+
+[source]
+----
+IPPRIO_MASK = 2^(mipreemptcfg.preemptmsk) - 1
+ipprio[i]   = iprio[i] & IPPRIO_MASK
+----
+
+A nested horizontal preemption will happen if and only if an interrupt i is pending and enabled,
+and the preemption priority of that interrupt ipprio[i] excceds (i.e. is numerically smaller) than {mintstatus}.{mipprio}.

```

Comment:
Just to make sure: this is essentially a renaming, and the existing functionality of mulitple interrupt levels and preemption is preserved?

Reply:
yes. it is going back to how previous versions of the spec worked, but using preemption priority instead of level.
In contrast to level, for the preemption priority, a smaller value has priority over a small value, to make it consistent with AIA and any other RISC-V spec.